scripts/quickstart

```diff
@@ -1,94 +1,145 @@
+
+  echo_bold "==> Setting up environment..."
+  # shellcheck source=scripts/make-env
+  cat >"${CONFIG_DIR}/activate" <(source "${DIR}/make-env")
+
+  echo_bold "==> Adding default compose file..."
+  cp "${BASE_DIR}/compose/template.yml" "${CONFIG_DIR}/docker-compose.yml"
+
+  if [ "${patch_hosts}" = "true" ]; then
+    echo_bold "==> Patching /etc/hosts..."
+    # shellcheck source=scripts/patch-hosts
+    source "${DIR}/patch-hosts" "${DOMAIN}"
+  fi
+
+  echo_bold "==> Success!"
+  echo '  - Start the instance with: ./scripts/compose up -d'
+  echo '  - Stop the instance with: ./scripts/compose stop'
+  echo '  - To create the superuser, see: ./scripts/create-superuser -h'
+  echo "  - Use the following certificate with Balena CLI: ${CERTS_DIR}/root/ca.crt"
 
-show_help=false
-patch_hosts=false
-while getopts ":hpd:U:P:" opt; do
-  case "${opt}" in
-    h) show_help=true;;
-    p) patch_hosts=true;;
-    d) DOMAIN="${OPTARG}";;
-    U) SUPERUSER_EMAIL="${OPTARG}";;
-    P) SUPERUSER_PASSWORD="${OPTARG}";;
-    *)
-      echo "Invalid argument: -${OPTARG}"
-      usage
-      exit 1
-      ;;
-  esac
-done
-shift $((OPTIND-1))
-
-if [ -z "${SUPERUSER_EMAIL}" ] || [ -z "${SUPERUSER_PASSWORD}" ]; then
-  usage
-  exit 1
-fi
-
-if [ "$show_help" = "true" ]; then
-  usage
-  exit 1
-fi
-
-echo_bold() {
-  printf "\\033[1m%s\\033[0m\\n" "${@}"
 }
 
-if [ -d "$CONFIG_DIR" ]; then
-  echo 'Configuration directory already exists; please remove it first.'
-  exit 1
-fi
-
-echo_bold "==> Creating new configuration at: $CONFIG_DIR"
-mkdir -p "$CONFIG_DIR" "$CERTS_DIR"
-
-echo_bold "==> Generating root CA cert..."
-# shellcheck source=scripts/gen-root-ca
-source "${DIR}/gen-root-ca" "${DOMAIN}" "${CERTS_DIR}"
-
-echo_bold "==> Generating root cert chain for haproxy..."
-# shellcheck source=scripts/gen-root-cert
-source "${DIR}/gen-root-cert" "${DOMAIN}" "${CERTS_DIR}"
-
-echo_bold "==> Generating token auth cert..."
-# shellcheck source=scripts/gen-token-auth-cert
-source "${DIR}/gen-token-auth-cert" "${DOMAIN}" "${CERTS_DIR}"
-
-echo_bold "==> Generating VPN CA, cert and dhparam (this may take a while)..."
-# shellcheck source=scripts/gen-vpn-certs
-source "${DIR}/gen-vpn-certs" "${DOMAIN}" "${CERTS_DIR}"
-
-echo_bold "==> Setting up environment..."
-# shellcheck source=scripts/make-env
-cat >"${CONFIG_DIR}/activate" <(source "${DIR}/make-env")
-
-echo_bold "==> Adding default compose file..."
-cp "${BASE_DIR}/compose/template.yml" "${CONFIG_DIR}/docker-compose.yml"
-
-if [ "${patch_hosts}" = "true" ]; then
-  echo_bold "==> Patching /etc/hosts..."
-  # shellcheck source=scripts/patch-hosts
-  source "${DIR}/patch-hosts" "${DOMAIN}"
-fi
-
-echo_bold "==> Success!"
-echo '  - Start the instance with: ./scripts/compose up -d'
-echo '  - Stop the instance with: ./scripts/compose stop'
-echo '  - To create the superuser, see: ./scripts/create-superuser -h'
-echo "  - Use the following certificate with Balena CLI: ${CERTS_DIR}/root/ca.crt"
+quickstart "$@"
```

Comment:
How come you've wrapped the script in a function?

Reply:
good practice; it makes sure the entire script is in-tact prior to running anything

Reply:
I believe it is considered good practice when the script is being streamed via curl (allá `curl | bash`), but it isn't necessary here.